rsconnect/api.py

```diff
@@ -292,7 +292,13 @@ def get_content(self, content_guid):
         return results
 
     def wait_for_task(
-        self, task_id, log_callback, abort_func=lambda: False, timeout=get_timeout(), poll_wait=0.5, raise_on_error=True
+        self,
```

Comment:
The `ending` calculation can overflow.

```
>>> t = time.time()
>>> ms = sys.maxsize
>>> ms
9223372036854775807
>>> t
1688655596.327841
>>> t + ms
9.223372038543432e+18
```

Should we switch the logic that computes `ending` into something similar to what is in `wait_until_task_is_successful()`, where we `time.time() - start_time < timeout`? This approach avoids using values greater than `sys.maxsize`.

Reply:
Thanks, @aronatkins - I adjusted the logic to avoid the overflow error.

https://github.com/rstudio/rsconnect-python/pull/445/commits/456c6914636768f0ca69b124e01847ee9d07720a